Job: Docker-PRs-gccgo FAILED
Here are what we parsed from the logs at https://jenkins.dockerproject.org/job/Docker-PRs-gccgo/1667/consoleText as the failures:

~~~console

---
###################### 100.0%
######################################################################## 100.0%

Download of images into 'bundles/1.11.0-dev/test-integration-cli/frozen-images' complete.
Use something like the following to load the result into a Docker daemon:
  tar -cC 'bundles/1.11.0-dev/test-integration-cli/frozen-images' . | docker load
++++ tar -cC bundles/1.11.0-dev/test-integration-cli/frozen-images .
++++ docker load
++++ set +x
Untagged: hello-world:latest
---> Making bundle: .ensure-httpserver (in bundles/1.11.0-dev/test-integration-cli)
---> Making bundle: .ensure-syscall-test (in bundles/1.11.0-dev/test-integration-cli)
+ go test -test.timeout=180m -check.v github.com/docker/docker/integration-cli
# github.com/docker/docker/integration-cli
./docker_cli_daemon_test.go:190:38: error: argument 1 has incompatible type (cannot use type int64 as type int)
  newBasesize, err := convertBasesize(newBasesizeBytes)
                                      ^
FAIL	github.com/docker/docker/integration-cli [build failed]
---> Making bundle: .integration-daemon-stop (in bundles/1.11.0-dev/test-integration-cli)
+++++ cat bundles/1.11.0-dev/test-integration-cli/docker.pid
++++ kill 5313
++++ /etc/init.d/apparmor stop
Clearing AppArmor profiles cache:.
All profile caches have been cleared, but no profiles have been unloaded.
Unloading profiles will leave already running processes permanently
unconfined, which can lead to unexpected situations.

To set a process to complain mode, use the command line tool
'aa-complain'. To really tear down all profiles, run the init script
with the 'teardown' option."
Build step 'Execute shell' marked build as failure
[PostBuildScript] - Execution post build scripts.
[Docker-PRs-gccgo] $ /bin/sh -xe /tmp/hudson8010305953727496379.sh
+ set +x


---
Now starting POST-BUILD steps
---

Error response from daemon: no such id: docker-pr-gccgo1667
Error: failed to remove containers: [docker-pr-gccgo1667]
Notifying endpoint 'HTTP:https://leeroy.dockerproject.org/notifi
---

~~~

--- a/integration-cli/docker_cli_daemon_test.go
+++ b/integration-cli/docker_cli_daemon_test.go
@@ -20,6 +20,7 @@ import (
 	"time"
 
 	"github.com/docker/docker/pkg/integration/checker"
+	"github.com/docker/go-units"
 	"github.com/docker/libnetwork/iptables"
 	"github.com/docker/libtrust"
 	"github.com/go-check/check"
@@ -152,6 +153,44 @@ func (s *DockerDaemonSuite) TestDaemonStartIptablesFalse(c *check.C) {
 	if err := s.d.Start("--iptables=false"); err != nil {
 		c.Fatalf("we should have been able to start the daemon with passing iptables=false: %v", err)
 	}
+}
+
+// Make sure we cannot shrink base device at daemon restart.
+func (s *DockerDaemonSuite) TestDaemonRestartWithInvalidBasesize(c *check.C) {
+	testRequires(c, Devicemapper)
+	c.Assert(s.d.Start(), check.IsNil)
+
+	oldBasesizeBytes := s.d.getBaseDeviceSize(c)
+	var newBasesizeBytes int64 = 1073741824 //1GB in bytes
+
+	if newBasesizeBytes < oldBasesizeBytes {
+		err := s.d.Restart("--storage-opt", fmt.Sprintf("dm.basesize=%d", newBasesizeBytes))
+		c.Assert(err, check.IsNil, check.Commentf("daemon should not have started as new base device size is less than existing base device size: %v", err))
+	}
+	c.Assert(s.d.Stop(), check.IsNil)
+}
+
+// Make sure we can grow base device at daemon restart.
+func (s *DockerDaemonSuite) TestDaemonRestartWithIncreasedBasesize(c *check.C) {
+	testRequires(c, Devicemapper)
+	c.Assert(s.d.Start(), check.IsNil)
+
+	oldBasesizeBytes := s.d.getBaseDeviceSize(c)
+
+	var newBasesizeBytes int64 = 53687091200 //50GB in bytes
+
+	if newBasesizeBytes < oldBasesizeBytes {
+		c.Skip(fmt.Sprintf("New base device size (%v) must be greater than (%s)", units.HumanSize(float64(newBasesizeBytes)), units.HumanSize(float64(oldBasesizeBytes))))
+	}
+
+	err := s.d.Restart("--storage-opt", fmt.Sprintf("dm.basesize=%d", newBasesizeBytes))
+	c.Assert(err, check.IsNil, check.Commentf("we should have been able to start the daemon with increased base device size: %v", err))
+
+	basesizeAfterRestart := s.d.getBaseDeviceSize(c)
+	newBasesize, err := convertBasesize(newBasesizeBytes)
+	c.Assert(err, check.IsNil, check.Commentf("Error in converting base device size: %v", err))
+	c.Assert(newBasesize, check.Equals, basesizeAfterRestart, check.Commentf("Basesize passed is not equal to Basesize set"))
+	c.Assert(s.d.Stop(), check.IsNil)
 }
 
 // Issue #8444: If docker0 bridge is modified (intentionally or unintentionally) and

--- a/integration-cli/docker_utils.go
+++ b/integration-cli/docker_utils.go
@@ -25,11 +25,13 @@ import (
 	"github.com/docker/docker/opts"
 	"github.com/docker/docker/pkg/httputils"
 	"github.com/docker/docker/pkg/integration"
+	"github.com/docker/docker/pkg/integration/checker"
 	"github.com/docker/docker/pkg/ioutils"
 	"github.com/docker/docker/pkg/stringutils"
 	"github.com/docker/engine-api/types"
 	"github.com/docker/go-connections/sockets"
 	"github.com/docker/go-connections/tlsconfig"
+	"github.com/docker/go-units"
 	"github.com/go-check/check"
 )
 
@@ -461,6 +463,32 @@ func (d *Daemon) sock() string {
 func (d *Daemon) waitRun(contID string) error {
 	args := []string{"--host", d.sock()}
 	return waitInspectWithArgs(contID, "{{.State.Running}}", "true", 10*time.Second, args...)
+}
+
+func (d *Daemon) getBaseDeviceSize(c *check.C) int64 {
+
+	infoCmdOutput, _, err := runCommandPipelineWithOutput(
+		exec.Command(dockerBinary, "-H", d.sock(), "info"),
+		exec.Command("grep", "Base Device Size"),
+	)
+	c.Assert(err, checker.IsNil)
+	basesizeSlice := strings.Split(infoCmdOutput, ":")
+	basesize := strings.Trim(basesizeSlice[1], " ")
+	basesize = strings.Trim(basesize, "\n")[:len(basesize)-3]
+	basesizeFloat, err := strconv.ParseFloat(strings.Trim(basesize, " "), 64)
+	c.Assert(err, checker.IsNil)
+	basesizeBytes := int64(basesizeFloat) * (1024 * 1024 * 1024)
+	return basesizeBytes
+}
+
+func convertBasesize(basesizeBytes int64) (int64, error) {
+	basesize := units.HumanSize(float64(basesizeBytes))
+	basesize = strings.Trim(basesize, " ")[:len(basesize)-3]
+	basesizeFloat, err := strconv.ParseFloat(strings.Trim(basesize, " "), 64)
+	if err != nil {
+		return 0, err
+	}
+	return int64(basesizeFloat) * 1024 * 1024 * 1024, nil
 }
 
 // Cmd will execute a docker CLI command against this Daemon.

--- a/integration-cli/requirements.go
+++ b/integration-cli/requirements.go
@@ -103,6 +103,18 @@ var (
 		},
 		"Test requires underlying root filesystem not be backed by overlay.",
 	}
+
+	Devicemapper = testRequirement{
+		func() bool {
+			cmd := exec.Command("grep", "^devicemapper / devicemapper", "/proc/mounts")
+			if err := cmd.Run(); err != nil {
+				return false
+			}
+			return true
+		},
+		"Test requires underlying root filesystem to be backed by devicemapper.",
+	}
+
 	IPv6 = testRequirement{
 		func() bool {
 			cmd := exec.Command("test", "-f", "/proc/net/if_inet6")

